### PR TITLE
Better ledger version comparison

### PR DESCRIPTION
**Pull request by marceloneil**
Currently the ledger version comparison is a bit lacking. If the bitcoin ledger app was updated to 1.2.0, electrum would assume incompatibility with multi-output, segwit and native segwit.

Instead of adding any new dependencies for this simple task (ie. semver), I simply compare the versions as tuples.

Let me know if I should change anything.

### CODE REVIEW

File: plugins/ledger/ledger.py [hunk 1]
```diff
@@ -118,12 +123,12 @@ def supports_native_segwit(self):
...
-            self.multiOutputSupported = int(firmware[0]) >= 1 and int(firmware[1]) >= 1 and int(firmware[2]) >= 4
-            self.segwitSupported = (int(firmware[0]) >= 1 and int(firmware[1]) >= 1 and int(firmware[2]) >= 10) or (firmwareInfo['specialVersion'] == 0x20 and int(firmware[0]) == 1 and int(firmware[1]) == 0 and int(firmware[2]) >= 4)
-            self.nativeSegwitSupported = int(firmware[0]) >= 1 and int(firmware[1]) >= 1 and int(firmware[2]) >= 10
+            firmware = firmwareInfo['version']
+            self.multiOutputSupported = self.versiontuple(firmware) >= self.versiontuple(MULTI_OUTPUT_SUPPORT)
+            self.nativeSegwitSupported = self.versiontuple(firmware) >= self.versiontuple(SEGWIT_SUPPORT)
+            self.segwitSupported = self.nativeSegwitSupported or (firmwareInfo['specialVersion'] == 0x20 and self.multiOutputSupported)
 
-            if not checkFirmware(firmware):
+            if not checkFirmware(firmwareInfo):
```
Comment (SomberNight) [inline plugins/ledger/ledger.py]: Why? Looks like this was `firmwareInfo['version'].split(".")`.
Though it seems that function just returns True anyway: https://github.com/LedgerHQ/btchip-python/blob/c2bb2166cdaec95795517811c6f6e16038401f1d/btchip/btchipFirmwareWizard.py#L20-L21

File: plugins/ledger/ledger.py [hunk 2]
```diff
@@ -118,12 +123,12 @@ def supports_native_segwit(self):
...
         try:
             firmwareInfo = self.dongleObject.getFirmwareVersion()
-            firmware = firmwareInfo['version'].split(".")
-            self.multiOutputSupported = int(firmware[0]) >= 1 and int(firmware[1]) >= 1 and int(firmware[2]) >= 4
-            self.segwitSupported = (int(firmware[0]) >= 1 and int(firmware[1]) >= 1 and int(firmware[2]) >= 10) or (firmwareInfo['specialVersion'] == 0x20 and int(firmware[0]) == 1 and int(firmware[1]) == 0 and int(firmware[2]) >= 4)
-            self.nativeSegwitSupported = int(firmware[0]) >= 1 and int(firmware[1]) >= 1 and int(firmware[2]) >= 10
+            firmware = firmwareInfo['version']
+            self.multiOutputSupported = self.versiontuple(firmware) >= self.versiontuple(MULTI_OUTPUT_SUPPORT)
+            self.nativeSegwitSupported = self.versiontuple(firmware) >= self.versiontuple(SEGWIT_SUPPORT)
+            self.segwitSupported = self.nativeSegwitSupported or (firmwareInfo['specialVersion'] == 0x20 and self.multiOutputSupported)
```
Comment (SomberNight) [inline plugins/ledger/ledger.py]: `self.multiOutputSupported` is `1.1.4` but before this change this was `1.0.4`. Is this change intentional?

Comment (SomberNight) [reply plugins/ledger/ledger.py]: I mean regarding the second branch of `self.segwitSupported`

File: plugins/ledger/ledger.py (hunk 1 above)
Comment (marceloneil) [reply plugins/ledger/ledger.py]: In the event that btchip implements the `checkFirmware` function, I'm assuming that they would take the firmware value in the same format that btchip-python returns. Do you think that I should revert it?

File: plugins/ledger/ledger.py (hunk 2 above)
Comment (marceloneil) [reply plugins/ledger/ledger.py]: Oh no! You're totally correct, I missed that. Good catch!

**Comment by marceloneil** [comment]
It might be better to have a `compare_version(v1, v2)` function, it would save some space, thoughts?

**Comment by SomberNight** [comment]
Thanks, I think this is fine as is.
